### PR TITLE
chore: add menu recommendation types

### PR DESCRIPTION
**Pull request by SicpamaMinwoo**
Add some types for menu recommendation system

### CODE REVIEW

File: types/menu.type.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+export type ItemBaseRecommendMenuUnit = {
```
Comment (hewison-chris) [inline types/menu.type.ts:1]:
```suggestion
export type ItemBasedRecommendMenuUnit = {
```

File: types/menu.type.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  precedentMenuIds: number[];
+  subsequentMenuIds: number[];
+};
+
+export type ItemBaseRecommendMenuStructure = {
```
Comment (hewison-chris) [inline types/menu.type.ts:6]:
```suggestion
export type ItemBasedRecommendMenuStructure = {
```

File: types/menu.type.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  sequence: number;
+  defaultMenuIds: number[];
+  recommends: ItemBaseRecommendMenuUnit[];
```
Comment (hewison-chris) [inline types/menu.type.ts:9]:
```suggestion
  recommends: ItemBasedRecommendMenuUnit[];
```

File: types/menu.type.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+};
+
+export type UserBaseRecommendMenuUnit = {
```
Comment (hewison-chris) [inline types/menu.type.ts:12]:
```suggestion
export type UserBasedRecommendMenuUnit = {
```

File: types/menu.type.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  menuId: number;
+  probability: number;
+};
+
+export type UserBaseRecommendMenuStructure = {
```
Comment (hewison-chris) [inline types/menu.type.ts:17]:
```suggestion
export type UserBasedRecommendMenuStructure = {
```

File: types/menu.type.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  totalUserNumber: number;
+  recommendMenus: UserBaseRecommendMenuUnit[];
```
Comment (hewison-chris) [inline types/menu.type.ts:19]:
```suggestion
  recommendMenus: UserBasedRecommendMenuUnit[];
```

File: types/menu.type.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+};
+
+export type UpdateItemBaseRecommendMenuRequest = {
```
Comment (hewison-chris) [inline types/menu.type.ts:22]:
```suggestion
export type UpdateItemBasedRecommendMenuRequest = {
```

File: types/menu.type.ts [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  menuId: number;
+  data: ItemBaseRecommendMenuStructure[];
```
Comment (hewison-chris) [inline types/menu.type.ts:24]:
```suggestion
  data: ItemBasedRecommendMenuStructure[];
```

File: types/menu.type.ts [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+};
+
+export type UpdateRecommendMenusRequest = {
+  itemBase?: UpdateItemBaseRecommendMenuRequest[];
```
Comment (hewison-chris) [inline types/menu.type.ts:28]:
```suggestion
  itemBase?: UpdateItemBasedRecommendMenuRequest[];
```

File: types/menu.type.ts [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+  userBase?: UserBaseRecommendMenuStructure[];
```
Comment (hewison-chris) [inline types/menu.type.ts:29]:
```suggestion
  userBase?: UserBasedRecommendMenuStructure[];
```

**Review by hewison-chris — CHANGES_REQUESTED** [review]
A few updates to names

**Comment by SicpamaMinwoo** [comment]
I renamed types @hewison-chris

**Review by hewison-chris — APPROVED** [review]
LGTM

**Pull request merged**